presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/MapColumnValidator.java

```diff
@@ -0,0 +1,108 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.checksum;
+
+import com.facebook.presto.spi.type.MapType;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.tree.CoalesceExpression;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.LongLiteral;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.SingleColumn;
+import com.facebook.presto.verifier.framework.Column;
+import com.google.common.collect.ImmutableList;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.facebook.presto.verifier.framework.VerifierUtil.delimitedIdentifier;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public class MapColumnValidator
+        implements ColumnValidator
+{
+    @Inject
+    public MapColumnValidator()
+    {
+    }
+
+    @Override
+    public List<SingleColumn> generateChecksumColumns(Column column)
+    {
+        checkArgument(column.getType() instanceof MapType, "Expect MapType, found %s", column.getType().getDisplayName());
+        Type keyType = ((MapType) column.getType()).getKeyType();
+        Type valueType = ((MapType) column.getType()).getValueType();
+
+        ImmutableList.Builder<Expression> checksums = ImmutableList.builder();
+        checksums.add(new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column.getIdentifier())));
+
+        Expression mapCardinalitySum = new CoalesceExpression(
+                new FunctionCall(
+                        QualifiedName.of("sum"),
+                        ImmutableList.of(new FunctionCall(QualifiedName.of("cardinality"), ImmutableList.of(column.getIdentifier())))),
+                new LongLiteral("0"));
+
+        if (keyType.isOrderable()) {
+            FunctionCall mapKeys = new FunctionCall(QualifiedName.of("map_keys"), ImmutableList.of(column.getIdentifier()));
+            FunctionCall sortKeys = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(mapKeys));
+            checksums.add(new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(sortKeys)));
+        }
+
+        if (valueType.isOrderable()) {
+            FunctionCall mapValues = new FunctionCall(QualifiedName.of("map_values"), ImmutableList.of(column.getIdentifier()));
+            FunctionCall sortValues = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(mapValues));
+            checksums.add(new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(sortValues)));
+        }
+
+        return ImmutableList.of(
+                new SingleColumn(new CoalesceExpression(checksums.build()), Optional.of(delimitedIdentifier(getChecksumColumnAlias(column)))),
```

Comment:
This doesn't seem right. I think we need to have multiple columns here, e.g. checksum(m), checksum(array_sort(map_keys(m))) and checksum(array_sort(map_values(m)))